dtests/src/test/java/io/snappydata/hydra/cdcConnector/SnappyCDCTest.java

```diff
@@ -1,41 +1,359 @@
+    List hostList = getHostNames();
+    if (orgName.renameTo(bkName)) {
+      Log.getLogWriter().info("File renamed to " + bkName);
+    } else {
+      Log.getLogWriter().info("Error");
+    }
+    try {
+      File tempConfFile = new File(snappyPath + "/conf/" + nodeType);
+      FileWriter fw = new FileWriter(tempConfFile, true);
+      Log.getLogWriter().info("The hostLost size = " + hostList.size());
+      for (int i = 0; i < hostList.size(); i++) {
+        String pidString = String.valueOf(hostList.get(i));
+        Log.getLogWriter().info("pidString : " + pidString);
+        int pid = Integer.parseInt(pidString);
+        Log.getLogWriter().info("Server Pid is : " + pidString);
+        String pidHost = snappyTest.getPidHost(Integer.toString(pid));
+        String newConfig = pidHost + " " + nodeConfig + "/" + pidHost + "  -critical-heap-percentage=95 \n";
+        fw.write(newConfig);
+      }
+      fw.close();
+    } catch (IOException ex) {
+      Log.getLogWriter().info("Caught exception in clusterModifyAndRestart " + ex.getMessage());
+    }
+  }
+
+  public void clusterRestart(String snappyPath, Boolean isStopStart, String nodeType, Boolean isBackupRecovery, String nodeConfig, Boolean isModifyConf) {
+    try {
+      File log = new File(".");
+      String dest = log.getCanonicalPath() + File.separator + "clusterRestart.log";
+      Log.getLogWriter().info("The destination file is " + dest);
+      File logFile = new File(dest);
+      String dirPath = SnappyCDCPrms.getDataLocation();
+      //Stop cluster
+      if (isStopStart) {
+        stopCluster(snappyPath, logFile);
+      }
+      if (nodeType.equalsIgnoreCase("leads")) {
+        //start locator:
+        ProcessBuilder pb = new ProcessBuilder(snappyPath + "/sbin/snappy-locators.sh", "start");
+        snappyTest.executeProcess(pb, logFile);
+        //start leader:
+        ProcessBuilder pb1 = new ProcessBuilder(snappyPath + "/sbin/snappy-leads.sh", "start");
+        snappyTest.executeProcess(pb1, logFile);
+      }
+      if (nodeType.equalsIgnoreCase("servers")) {
+        //only start servers first:
+        ProcessBuilder pb1 = new ProcessBuilder(snappyPath + "/sbin/snappy-servers.sh", "start");
+        snappyTest.executeProcess(pb1, logFile);
+      }
+      if (isBackupRecovery) {
+        Log.getLogWriter().info("Inside isBackupRecovery true loop");
+        removeDiskStoreFiles(dirPath);
+      }
+      if (isModifyConf) {
+        clusterModifyAndRestart(snappyPath, nodeType, nodeConfig);
+      }
+      //Start the cluster after 1 min
+      // Thread.sleep(60000);
+      Log.getLogWriter().info("Invoking startCluster");
+      startCluster(snappyPath, logFile, isModifyConf);
+      snappyTest.recordSnappyProcessIDinNukeRun("ServerLauncher");
+    } catch (GemFireConfigException ex) {
+      Log.getLogWriter().info("Got the expected exception when starting a new server ,without locators " + ex.getMessage());
```

Comment:
Make the logging generic. Mention about servers and not just the new server node.